### ISSUE TITLE
use-focus-trap does not work with Select

### ISSUE THREAD

**Issue by skelawsky** [mantinedev/mantine#1602]
### What package has an issue

@mantine/core

### Describe the bug

Focus trap with Select does not work as expected, stops at last dom element and does not come back to Select. 
Without Select you can hold TAB key to loop through all elemets

```
<div ref={focusTrapRef}>
        <Select data={[]} />
        <NumberInput></NumberInput>
        <NumberInput></NumberInput>
</div>
```

### In which browser did the problem occur

Chrome, Brave

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/adoring-ganguly-kvydd8?file=/src/App.js

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

No

**Comment by skelawsky** [comment]
I added codesandbox.

**Comment by wes337** [comment]
This is happening because the Select adds a hidden element, and the focus selector in the use-focus-trap hook does not account for hidden elements. This also should apply to disabled elements
